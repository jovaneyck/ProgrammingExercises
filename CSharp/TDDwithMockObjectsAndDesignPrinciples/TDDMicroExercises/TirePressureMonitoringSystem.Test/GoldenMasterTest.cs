using System;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace TDDMicroExercises.TirePressureMonitoringSystem.Test
{
    [TestFixture]
    public class GoldenMasterTest
    {
        [Test]
        public void NUnitSanityCheck()
        {
            Assert.IsTrue(true);
        }

        [Test]
        [UseReporter(typeof(NUnitReporter))]
        public void AlarmUsedInRealisticScenario()
        {
            //Note that this is a VERY high-level regression test: 
            // the dummy sensor always returns the same value, so the alarm will always be on after the first pressure check.
            StringBuilder s = new StringBuilder();
            for(int i = 0; i < 100; i++)
            {
                Alarm a = new Alarm();
                for(int j = 0; j < 10; j++)
                {
                    s.AppendLine(String.Format("Run {0} {1}: {2}", i,j, a.AlarmOn));
                    a.Check();
                }
            }

            Approvals.Verify(s);
        }
    }
}
