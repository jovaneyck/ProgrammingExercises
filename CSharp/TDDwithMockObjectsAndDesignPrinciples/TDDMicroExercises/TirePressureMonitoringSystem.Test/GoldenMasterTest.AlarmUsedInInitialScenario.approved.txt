Run 0 0: False
Run 0 1: True
Run 0 2: True
Run 0 3: True
Run 0 4: True
Run 0 5: True
Run 0 6: True
Run 0 7: True
Run 0 8: True
Run 0 9: True
Run 1 0: False
Run 1 1: True
Run 1 2: True
Run 1 3: True
Run 1 4: True
Run 1 5: True
Run 1 6: True
Run 1 7: True
Run 1 8: True
Run 1 9: True
Run 2 0: False
Run 2 1: True
Run 2 2: True
Run 2 3: True
Run 2 4: True
Run 2 5: True
Run 2 6: True
Run 2 7: True
Run 2 8: True
Run 2 9: True
Run 3 0: False
Run 3 1: True
Run 3 2: True
Run 3 3: True
Run 3 4: True
Run 3 5: True
Run 3 6: True
Run 3 7: True
Run 3 8: True
Run 3 9: True
Run 4 0: False
Run 4 1: True
Run 4 2: True
Run 4 3: True
Run 4 4: True
Run 4 5: True
Run 4 6: True
Run 4 7: True
Run 4 8: True
Run 4 9: True
Run 5 0: False
Run 5 1: True
Run 5 2: True
Run 5 3: True
Run 5 4: True
Run 5 5: True
Run 5 6: True
Run 5 7: True
Run 5 8: True
Run 5 9: True
Run 6 0: False
Run 6 1: True
Run 6 2: True
Run 6 3: True
Run 6 4: True
Run 6 5: True
Run 6 6: True
Run 6 7: True
Run 6 8: True
Run 6 9: True
Run 7 0: False
Run 7 1: True
Run 7 2: True
Run 7 3: True
Run 7 4: True
Run 7 5: True
Run 7 6: True
Run 7 7: True
Run 7 8: True
Run 7 9: True
Run 8 0: False
Run 8 1: True
Run 8 2: True
Run 8 3: True
Run 8 4: True
Run 8 5: True
Run 8 6: True
Run 8 7: True
Run 8 8: True
Run 8 9: True
Run 9 0: False
Run 9 1: True
Run 9 2: True
Run 9 3: True
Run 9 4: True
Run 9 5: True
Run 9 6: True
Run 9 7: True
Run 9 8: True
Run 9 9: True
Run 10 0: False
Run 10 1: True
Run 10 2: True
Run 10 3: True
Run 10 4: True
Run 10 5: True
Run 10 6: True
Run 10 7: True
Run 10 8: True
Run 10 9: True
Run 11 0: False
Run 11 1: True
Run 11 2: True
Run 11 3: True
Run 11 4: True
Run 11 5: True
Run 11 6: True
Run 11 7: True
Run 11 8: True
Run 11 9: True
Run 12 0: False
Run 12 1: True
Run 12 2: True
Run 12 3: True
Run 12 4: True
Run 12 5: True
Run 12 6: True
Run 12 7: True
Run 12 8: True
Run 12 9: True
Run 13 0: False
Run 13 1: True
Run 13 2: True
Run 13 3: True
Run 13 4: True
Run 13 5: True
Run 13 6: True
Run 13 7: True
Run 13 8: True
Run 13 9: True
Run 14 0: False
Run 14 1: True
Run 14 2: True
Run 14 3: True
Run 14 4: True
Run 14 5: True
Run 14 6: True
Run 14 7: True
Run 14 8: True
Run 14 9: True
Run 15 0: False
Run 15 1: True
Run 15 2: True
Run 15 3: True
Run 15 4: True
Run 15 5: True
Run 15 6: True
Run 15 7: True
Run 15 8: True
Run 15 9: True
Run 16 0: False
Run 16 1: True
Run 16 2: True
Run 16 3: True
Run 16 4: True
Run 16 5: True
Run 16 6: True
Run 16 7: True
Run 16 8: True
Run 16 9: True
Run 17 0: False
Run 17 1: True
Run 17 2: True
Run 17 3: True
Run 17 4: True
Run 17 5: True
Run 17 6: True
Run 17 7: True
Run 17 8: True
Run 17 9: True
Run 18 0: False
Run 18 1: True
Run 18 2: True
Run 18 3: True
Run 18 4: True
Run 18 5: True
Run 18 6: True
Run 18 7: True
Run 18 8: True
Run 18 9: True
Run 19 0: False
Run 19 1: True
Run 19 2: True
Run 19 3: True
Run 19 4: True
Run 19 5: True
Run 19 6: True
Run 19 7: True
Run 19 8: True
Run 19 9: True
Run 20 0: False
Run 20 1: True
Run 20 2: True
Run 20 3: True
Run 20 4: True
Run 20 5: True
Run 20 6: True
Run 20 7: True
Run 20 8: True
Run 20 9: True
Run 21 0: False
Run 21 1: True
Run 21 2: True
Run 21 3: True
Run 21 4: True
Run 21 5: True
Run 21 6: True
Run 21 7: True
Run 21 8: True
Run 21 9: True
Run 22 0: False
Run 22 1: True
Run 22 2: True
Run 22 3: True
Run 22 4: True
Run 22 5: True
Run 22 6: True
Run 22 7: True
Run 22 8: True
Run 22 9: True
Run 23 0: False
Run 23 1: True
Run 23 2: True
Run 23 3: True
Run 23 4: True
Run 23 5: True
Run 23 6: True
Run 23 7: True
Run 23 8: True
Run 23 9: True
Run 24 0: False
Run 24 1: True
Run 24 2: True
Run 24 3: True
Run 24 4: True
Run 24 5: True
Run 24 6: True
Run 24 7: True
Run 24 8: True
Run 24 9: True
Run 25 0: False
Run 25 1: True
Run 25 2: True
Run 25 3: True
Run 25 4: True
Run 25 5: True
Run 25 6: True
Run 25 7: True
Run 25 8: True
Run 25 9: True
Run 26 0: False
Run 26 1: True
Run 26 2: True
Run 26 3: True
Run 26 4: True
Run 26 5: True
Run 26 6: True
Run 26 7: True
Run 26 8: True
Run 26 9: True
Run 27 0: False
Run 27 1: True
Run 27 2: True
Run 27 3: True
Run 27 4: True
Run 27 5: True
Run 27 6: True
Run 27 7: True
Run 27 8: True
Run 27 9: True
Run 28 0: False
Run 28 1: True
Run 28 2: True
Run 28 3: True
Run 28 4: True
Run 28 5: True
Run 28 6: True
Run 28 7: True
Run 28 8: True
Run 28 9: True
Run 29 0: False
Run 29 1: True
Run 29 2: True
Run 29 3: True
Run 29 4: True
Run 29 5: True
Run 29 6: True
Run 29 7: True
Run 29 8: True
Run 29 9: True
Run 30 0: False
Run 30 1: True
Run 30 2: True
Run 30 3: True
Run 30 4: True
Run 30 5: True
Run 30 6: True
Run 30 7: True
Run 30 8: True
Run 30 9: True
Run 31 0: False
Run 31 1: True
Run 31 2: True
Run 31 3: True
Run 31 4: True
Run 31 5: True
Run 31 6: True
Run 31 7: True
Run 31 8: True
Run 31 9: True
Run 32 0: False
Run 32 1: True
Run 32 2: True
Run 32 3: True
Run 32 4: True
Run 32 5: True
Run 32 6: True
Run 32 7: True
Run 32 8: True
Run 32 9: True
Run 33 0: False
Run 33 1: True
Run 33 2: True
Run 33 3: True
Run 33 4: True
Run 33 5: True
Run 33 6: True
Run 33 7: True
Run 33 8: True
Run 33 9: True
Run 34 0: False
Run 34 1: True
Run 34 2: True
Run 34 3: True
Run 34 4: True
Run 34 5: True
Run 34 6: True
Run 34 7: True
Run 34 8: True
Run 34 9: True
Run 35 0: False
Run 35 1: True
Run 35 2: True
Run 35 3: True
Run 35 4: True
Run 35 5: True
Run 35 6: True
Run 35 7: True
Run 35 8: True
Run 35 9: True
Run 36 0: False
Run 36 1: True
Run 36 2: True
Run 36 3: True
Run 36 4: True
Run 36 5: True
Run 36 6: True
Run 36 7: True
Run 36 8: True
Run 36 9: True
Run 37 0: False
Run 37 1: True
Run 37 2: True
Run 37 3: True
Run 37 4: True
Run 37 5: True
Run 37 6: True
Run 37 7: True
Run 37 8: True
Run 37 9: True
Run 38 0: False
Run 38 1: True
Run 38 2: True
Run 38 3: True
Run 38 4: True
Run 38 5: True
Run 38 6: True
Run 38 7: True
Run 38 8: True
Run 38 9: True
Run 39 0: False
Run 39 1: True
Run 39 2: True
Run 39 3: True
Run 39 4: True
Run 39 5: True
Run 39 6: True
Run 39 7: True
Run 39 8: True
Run 39 9: True
Run 40 0: False
Run 40 1: True
Run 40 2: True
Run 40 3: True
Run 40 4: True
Run 40 5: True
Run 40 6: True
Run 40 7: True
Run 40 8: True
Run 40 9: True
Run 41 0: False
Run 41 1: True
Run 41 2: True
Run 41 3: True
Run 41 4: True
Run 41 5: True
Run 41 6: True
Run 41 7: True
Run 41 8: True
Run 41 9: True
Run 42 0: False
Run 42 1: True
Run 42 2: True
Run 42 3: True
Run 42 4: True
Run 42 5: True
Run 42 6: True
Run 42 7: True
Run 42 8: True
Run 42 9: True
Run 43 0: False
Run 43 1: True
Run 43 2: True
Run 43 3: True
Run 43 4: True
Run 43 5: True
Run 43 6: True
Run 43 7: True
Run 43 8: True
Run 43 9: True
Run 44 0: False
Run 44 1: True
Run 44 2: True
Run 44 3: True
Run 44 4: True
Run 44 5: True
Run 44 6: True
Run 44 7: True
Run 44 8: True
Run 44 9: True
Run 45 0: False
Run 45 1: True
Run 45 2: True
Run 45 3: True
Run 45 4: True
Run 45 5: True
Run 45 6: True
Run 45 7: True
Run 45 8: True
Run 45 9: True
Run 46 0: False
Run 46 1: True
Run 46 2: True
Run 46 3: True
Run 46 4: True
Run 46 5: True
Run 46 6: True
Run 46 7: True
Run 46 8: True
Run 46 9: True
Run 47 0: False
Run 47 1: True
Run 47 2: True
Run 47 3: True
Run 47 4: True
Run 47 5: True
Run 47 6: True
Run 47 7: True
Run 47 8: True
Run 47 9: True
Run 48 0: False
Run 48 1: True
Run 48 2: True
Run 48 3: True
Run 48 4: True
Run 48 5: True
Run 48 6: True
Run 48 7: True
Run 48 8: True
Run 48 9: True
Run 49 0: False
Run 49 1: True
Run 49 2: True
Run 49 3: True
Run 49 4: True
Run 49 5: True
Run 49 6: True
Run 49 7: True
Run 49 8: True
Run 49 9: True
Run 50 0: False
Run 50 1: True
Run 50 2: True
Run 50 3: True
Run 50 4: True
Run 50 5: True
Run 50 6: True
Run 50 7: True
Run 50 8: True
Run 50 9: True
Run 51 0: False
Run 51 1: True
Run 51 2: True
Run 51 3: True
Run 51 4: True
Run 51 5: True
Run 51 6: True
Run 51 7: True
Run 51 8: True
Run 51 9: True
Run 52 0: False
Run 52 1: True
Run 52 2: True
Run 52 3: True
Run 52 4: True
Run 52 5: True
Run 52 6: True
Run 52 7: True
Run 52 8: True
Run 52 9: True
Run 53 0: False
Run 53 1: True
Run 53 2: True
Run 53 3: True
Run 53 4: True
Run 53 5: True
Run 53 6: True
Run 53 7: True
Run 53 8: True
Run 53 9: True
Run 54 0: False
Run 54 1: True
Run 54 2: True
Run 54 3: True
Run 54 4: True
Run 54 5: True
Run 54 6: True
Run 54 7: True
Run 54 8: True
Run 54 9: True
Run 55 0: False
Run 55 1: True
Run 55 2: True
Run 55 3: True
Run 55 4: True
Run 55 5: True
Run 55 6: True
Run 55 7: True
Run 55 8: True
Run 55 9: True
Run 56 0: False
Run 56 1: True
Run 56 2: True
Run 56 3: True
Run 56 4: True
Run 56 5: True
Run 56 6: True
Run 56 7: True
Run 56 8: True
Run 56 9: True
Run 57 0: False
Run 57 1: True
Run 57 2: True
Run 57 3: True
Run 57 4: True
Run 57 5: True
Run 57 6: True
Run 57 7: True
Run 57 8: True
Run 57 9: True
Run 58 0: False
Run 58 1: True
Run 58 2: True
Run 58 3: True
Run 58 4: True
Run 58 5: True
Run 58 6: True
Run 58 7: True
Run 58 8: True
Run 58 9: True
Run 59 0: False
Run 59 1: True
Run 59 2: True
Run 59 3: True
Run 59 4: True
Run 59 5: True
Run 59 6: True
Run 59 7: True
Run 59 8: True
Run 59 9: True
Run 60 0: False
Run 60 1: True
Run 60 2: True
Run 60 3: True
Run 60 4: True
Run 60 5: True
Run 60 6: True
Run 60 7: True
Run 60 8: True
Run 60 9: True
Run 61 0: False
Run 61 1: True
Run 61 2: True
Run 61 3: True
Run 61 4: True
Run 61 5: True
Run 61 6: True
Run 61 7: True
Run 61 8: True
Run 61 9: True
Run 62 0: False
Run 62 1: True
Run 62 2: True
Run 62 3: True
Run 62 4: True
Run 62 5: True
Run 62 6: True
Run 62 7: True
Run 62 8: True
Run 62 9: True
Run 63 0: False
Run 63 1: True
Run 63 2: True
Run 63 3: True
Run 63 4: True
Run 63 5: True
Run 63 6: True
Run 63 7: True
Run 63 8: True
Run 63 9: True
Run 64 0: False
Run 64 1: True
Run 64 2: True
Run 64 3: True
Run 64 4: True
Run 64 5: True
Run 64 6: True
Run 64 7: True
Run 64 8: True
Run 64 9: True
Run 65 0: False
Run 65 1: True
Run 65 2: True
Run 65 3: True
Run 65 4: True
Run 65 5: True
Run 65 6: True
Run 65 7: True
Run 65 8: True
Run 65 9: True
Run 66 0: False
Run 66 1: True
Run 66 2: True
Run 66 3: True
Run 66 4: True
Run 66 5: True
Run 66 6: True
Run 66 7: True
Run 66 8: True
Run 66 9: True
Run 67 0: False
Run 67 1: True
Run 67 2: True
Run 67 3: True
Run 67 4: True
Run 67 5: True
Run 67 6: True
Run 67 7: True
Run 67 8: True
Run 67 9: True
Run 68 0: False
Run 68 1: True
Run 68 2: True
Run 68 3: True
Run 68 4: True
Run 68 5: True
Run 68 6: True
Run 68 7: True
Run 68 8: True
Run 68 9: True
Run 69 0: False
Run 69 1: True
Run 69 2: True
Run 69 3: True
Run 69 4: True
Run 69 5: True
Run 69 6: True
Run 69 7: True
Run 69 8: True
Run 69 9: True
Run 70 0: False
Run 70 1: True
Run 70 2: True
Run 70 3: True
Run 70 4: True
Run 70 5: True
Run 70 6: True
Run 70 7: True
Run 70 8: True
Run 70 9: True
Run 71 0: False
Run 71 1: True
Run 71 2: True
Run 71 3: True
Run 71 4: True
Run 71 5: True
Run 71 6: True
Run 71 7: True
Run 71 8: True
Run 71 9: True
Run 72 0: False
Run 72 1: True
Run 72 2: True
Run 72 3: True
Run 72 4: True
Run 72 5: True
Run 72 6: True
Run 72 7: True
Run 72 8: True
Run 72 9: True
Run 73 0: False
Run 73 1: True
Run 73 2: True
Run 73 3: True
Run 73 4: True
Run 73 5: True
Run 73 6: True
Run 73 7: True
Run 73 8: True
Run 73 9: True
Run 74 0: False
Run 74 1: True
Run 74 2: True
Run 74 3: True
Run 74 4: True
Run 74 5: True
Run 74 6: True
Run 74 7: True
Run 74 8: True
Run 74 9: True
Run 75 0: False
Run 75 1: True
Run 75 2: True
Run 75 3: True
Run 75 4: True
Run 75 5: True
Run 75 6: True
Run 75 7: True
Run 75 8: True
Run 75 9: True
Run 76 0: False
Run 76 1: True
Run 76 2: True
Run 76 3: True
Run 76 4: True
Run 76 5: True
Run 76 6: True
Run 76 7: True
Run 76 8: True
Run 76 9: True
Run 77 0: False
Run 77 1: True
Run 77 2: True
Run 77 3: True
Run 77 4: True
Run 77 5: True
Run 77 6: True
Run 77 7: True
Run 77 8: True
Run 77 9: True
Run 78 0: False
Run 78 1: True
Run 78 2: True
Run 78 3: True
Run 78 4: True
Run 78 5: True
Run 78 6: True
Run 78 7: True
Run 78 8: True
Run 78 9: True
Run 79 0: False
Run 79 1: True
Run 79 2: True
Run 79 3: True
Run 79 4: True
Run 79 5: True
Run 79 6: True
Run 79 7: True
Run 79 8: True
Run 79 9: True
Run 80 0: False
Run 80 1: True
Run 80 2: True
Run 80 3: True
Run 80 4: True
Run 80 5: True
Run 80 6: True
Run 80 7: True
Run 80 8: True
Run 80 9: True
Run 81 0: False
Run 81 1: True
Run 81 2: True
Run 81 3: True
Run 81 4: True
Run 81 5: True
Run 81 6: True
Run 81 7: True
Run 81 8: True
Run 81 9: True
Run 82 0: False
Run 82 1: True
Run 82 2: True
Run 82 3: True
Run 82 4: True
Run 82 5: True
Run 82 6: True
Run 82 7: True
Run 82 8: True
Run 82 9: True
Run 83 0: False
Run 83 1: True
Run 83 2: True
Run 83 3: True
Run 83 4: True
Run 83 5: True
Run 83 6: True
Run 83 7: True
Run 83 8: True
Run 83 9: True
Run 84 0: False
Run 84 1: True
Run 84 2: True
Run 84 3: True
Run 84 4: True
Run 84 5: True
Run 84 6: True
Run 84 7: True
Run 84 8: True
Run 84 9: True
Run 85 0: False
Run 85 1: True
Run 85 2: True
Run 85 3: True
Run 85 4: True
Run 85 5: True
Run 85 6: True
Run 85 7: True
Run 85 8: True
Run 85 9: True
Run 86 0: False
Run 86 1: True
Run 86 2: True
Run 86 3: True
Run 86 4: True
Run 86 5: True
Run 86 6: True
Run 86 7: True
Run 86 8: True
Run 86 9: True
Run 87 0: False
Run 87 1: True
Run 87 2: True
Run 87 3: True
Run 87 4: True
Run 87 5: True
Run 87 6: True
Run 87 7: True
Run 87 8: True
Run 87 9: True
Run 88 0: False
Run 88 1: True
Run 88 2: True
Run 88 3: True
Run 88 4: True
Run 88 5: True
Run 88 6: True
Run 88 7: True
Run 88 8: True
Run 88 9: True
Run 89 0: False
Run 89 1: True
Run 89 2: True
Run 89 3: True
Run 89 4: True
Run 89 5: True
Run 89 6: True
Run 89 7: True
Run 89 8: True
Run 89 9: True
Run 90 0: False
Run 90 1: True
Run 90 2: True
Run 90 3: True
Run 90 4: True
Run 90 5: True
Run 90 6: True
Run 90 7: True
Run 90 8: True
Run 90 9: True
Run 91 0: False
Run 91 1: True
Run 91 2: True
Run 91 3: True
Run 91 4: True
Run 91 5: True
Run 91 6: True
Run 91 7: True
Run 91 8: True
Run 91 9: True
Run 92 0: False
Run 92 1: True
Run 92 2: True
Run 92 3: True
Run 92 4: True
Run 92 5: True
Run 92 6: True
Run 92 7: True
Run 92 8: True
Run 92 9: True
Run 93 0: False
Run 93 1: True
Run 93 2: True
Run 93 3: True
Run 93 4: True
Run 93 5: True
Run 93 6: True
Run 93 7: True
Run 93 8: True
Run 93 9: True
Run 94 0: False
Run 94 1: True
Run 94 2: True
Run 94 3: True
Run 94 4: True
Run 94 5: True
Run 94 6: True
Run 94 7: True
Run 94 8: True
Run 94 9: True
Run 95 0: False
Run 95 1: True
Run 95 2: True
Run 95 3: True
Run 95 4: True
Run 95 5: True
Run 95 6: True
Run 95 7: True
Run 95 8: True
Run 95 9: True
Run 96 0: False
Run 96 1: True
Run 96 2: True
Run 96 3: True
Run 96 4: True
Run 96 5: True
Run 96 6: True
Run 96 7: True
Run 96 8: True
Run 96 9: True
Run 97 0: False
Run 97 1: True
Run 97 2: True
Run 97 3: True
Run 97 4: True
Run 97 5: True
Run 97 6: True
Run 97 7: True
Run 97 8: True
Run 97 9: True
Run 98 0: False
Run 98 1: True
Run 98 2: True
Run 98 3: True
Run 98 4: True
Run 98 5: True
Run 98 6: True
Run 98 7: True
Run 98 8: True
Run 98 9: True
Run 99 0: False
Run 99 1: True
Run 99 2: True
Run 99 3: True
Run 99 4: True
Run 99 5: True
Run 99 6: True
Run 99 7: True
Run 99 8: True
Run 99 9: True
